onnxruntime/core/session/inference_session.cc

```diff
@@ -772,11 +772,9 @@ static bool ModelUseFP16Helper(const onnx::TypeProto& type_proto) {
   return false;
 }
 
-static bool ModelUseFP16(const onnx::ModelProto& model_proto) {
-  auto& graph = model_proto.graph();
-  auto& inputs = graph.input();
-  for (auto& input : inputs) {
-    if (input.has_name() && input.has_type() && ModelUseFP16Helper(input.type())) {
+static bool ModelUseFP16(const Graph& graph) {
+  for (auto& input : graph.GetInputs()) {
+    if (input->Exists() && ModelUseFP16Helper(*(input->TypeAsProto()))) {
```

Comment:
Can this make sure the input has type?

Reply:
Check for name and type is already done here : https://github.com/microsoft/onnxruntime/blob/a7541f97530b4d0668a5147460dc55cee96f03aa/onnxruntime/core/graph/graph.cc#L758